### PR TITLE
PMP-1064 | JAN-163

### PR DESCRIPTION
**Pull request by dtiwarATS**

### CODE REVIEW

File: assets/src/adaptivity/scripting.ts [hunk 1]
```diff
@@ -57,25 +57,15 @@ export const getExpressionStringForValue = (
...
     let actuallyAString = false;
+    const expressions = extractAllExpressionsFromText(val);
+    // A expression will not have a ';' inside it. So if there is a ';' inside it, it is CSS.
+    const isCSSString = expressions.filter((e) => e.includes(';'));
+    if (isCSSString?.length) {
+      actuallyAString = true;
+    }
     try {
       const testEnv = new Environment(env);
-      const evalResult = evalScript(`let foo = ${val};`, testEnv);
```
Comment (dtiwarATS) [inline assets/src/adaptivity/scripting.ts:62]: Removing all the previous CSS related fixes. We don't need them anymore. I have test few CSS related tickets and they we working properly.

File: assets/src/adaptivity/scripting.ts [hunk 2]
```diff
@@ -57,25 +57,15 @@ export const getExpressionStringForValue = (
...
-          );
-          //matchingCssElements !== null then it means it's a CSS string so set actuallyAString=true so that it can be wrapped in ""
-          if (matchingCssElements) {
-            actuallyAString = true;
-          }
-        } catch (e) {
-          actuallyAString = true;
-        }
-      }
+      evalScript(`let foo = ${val};`, testEnv);
```
Comment (bsparks) [inline assets/src/adaptivity/scripting.ts:68]: still we need to test `evalResult != null` for things besides CSS. also I think if we have determined via the CSS test above that `if (actuallyAString === true)` then we don't need to continue testing

Comment (dtiwarATS) [reply assets/src/adaptivity/scripting.ts]: Added changes for the second point. I am still not sure why we need the evalResult != null section because if it fails we already setting  "actuallyAString = true;" in catch section.

File: assets/src/adaptivity/scripting.ts [hunk 3]
```diff
@@ -57,30 +57,21 @@ export const getExpressionStringForValue = (
...
+    if (isCSSString?.length) {
       actuallyAString = true;
     }
-
+    if (!actuallyAString) {
+      try {
+        const testEnv = new Environment(env);
+        evalScript(`let foo = ${val};`, testEnv);
+      } catch (e) {
+        // if we have parsing error then we're guessing it's CSS
```
Comment (bsparks) [inline assets/src/adaptivity/scripting.ts:71]: I think if we're here we're no longer guessing it's CSS because of above, but have determined that whatever it is can't be evaluated by the scripting engine and so we're assuming its a string

File: assets/src/adaptivity/scripting.ts [hunk 4]
```diff
@@ -57,30 +57,21 @@ export const getExpressionStringForValue = (
...
+    // A expression will not have a ';' inside it. So if there is a ';' inside it, it is CSS.
+    const isCSSString = expressions.filter((e) => e.includes(';'));
+    if (isCSSString?.length) {
       actuallyAString = true;
     }
-
+    if (!actuallyAString) {
+      try {
+        const testEnv = new Environment(env);
+        evalScript(`let foo = ${val};`, testEnv);
```
Comment (bsparks) [inline assets/src/adaptivity/scripting.ts:69]: I think we still want to do
```
const testResult = evalScript(`let foo = ${val};`, testEnv);
if (testResult?.result !== null) {
  actuallyAString = true;
}
```
as well, because the catch is only for exceptions, or maybe we should be checking `testResult.result.message` exists?

Comment (dtiwarATS) [reply assets/src/adaptivity/scripting.ts]: Done

File: assets/src/adaptivity/scripting.ts [hunk 5]
```diff
@@ -57,30 +57,25 @@ export const getExpressionStringForValue = (
...
+    if (isCSSString?.length) {
+      actuallyAString = true;
+    }
+    if (!actuallyAString) {
+      try {
+        const testEnv = new Environment(env);
+        const testResult = evalScript(`let foo = ${val};`, testEnv);
+        if (testResult?.result !== null) {
+          //expression {stage.foo} + {stage.bar} was failling if we set actuallyAString= true
+          actuallyAString = expressions?.length ? false : true;
```
Comment (bsparks) [inline assets/src/adaptivity/scripting.ts:72]: hmm what about if it's `round(37.234)` ?

Comment (dtiwarATS) [reply assets/src/adaptivity/scripting.ts:72]: It works properly and gets evaluated to 37 so the testResult?.result is null for round(37.234)

**Review by bsparks — APPROVED** [review]
